c10/xpu/XPUFunctions.cpp

```diff
@@ -27,6 +28,22 @@ struct DevicePool {
   std::unique_ptr<sycl::context> context;
 } gDevicePool;
 
+void identifyDeviceType(
+    std::vector<sycl::device>& device_list,
+    std::vector<bool>& is_igpu,
+    std::vector<bool>& is_dgpu) {
+  for (const auto i : c10::irange(device_list.size())) {
+    const auto& device = device_list[i];
+    if (device.is_gpu()) {
+      if (device.get_info<sycl::info::device::host_unified_memory>()) {
```

Comment:
Here refer to https://github.com/intel/llvm/blob/551579126fddafc9224d68b52c84eb13ad50d615/sycl/include/sycl/reduction.hpp#L1374
and
https://github.com/intel/llvm/blob/551579126fddafc9224d68b52c84eb13ad50d615/sycl/test-e2e/USM/usm_leak_check.cpp#L52
etc.